src/CacheTower.Providers.Redis/Entities/RedisCacheEntry.cs

```diff
@@ -10,17 +9,20 @@ namespace CacheTower.Providers.Redis.Entities
 	/// </summary>
 	/// <typeparam name="T">The type of the cached value</typeparam>
 	[ProtoContract]
+	[DataContract]
 	public class RedisCacheEntry<T>
 	{
 		/// <summary>
 		/// The expiry date of the cache entry.
 		/// </summary>
 		[ProtoMember(1)]
+		[DataMember(Name = "expiry")]
 		public DateTime Expiry { get; set; }
 		/// <summary>
 		/// The cached value itself.
 		/// </summary>
 		[ProtoMember(2)]
+		[DataMember(Name = "value")]
 		public T? Value { get; set; }
 	}
```

Comment:
I think this class might cause us some issues in the future. The additional `DataContract` attributes should help with serialization via `Newtonsoft.Json` though will likely be a problem for any other serializers in the future. Like, at some point we will be removing the `Proto*` attributes and removing the dependency on protobuf directly but that will prevent it from working. (We'll have the same issue with the file manifest entries etc in that cache layer)

Don't know yet what the solution will be but probably a good idea to figure that out first.

Reply:
Ye I took a crack at solving for that but didn't push the changes. It seems like each ICacheSerializer would need to be able to construct objects of a type (such as ProtobufCacheEntry) that is decorated with serialization attributes only relevant to it.

I don't quite get how RedisCacheEntry, ManifestEntry and CacheEntry are supposed to fit together. Does CacheEntry still exist in this new world?

Reply:
So `CacheEntry` is the core entry type according to Cache Tower and is fundamentally what is used for memory caching. Keeping that going forward is important.

With file-based caching, I store the data separately from the "manifest" - that is to say, there is one file that has an index of all the other files with expiry dates. That's why you see a `ManifestEntry` type with a matching interface. This interface worked out well with Protobuf serialization for files implementing its own `IManifestEntry`, allowing me to mark-up the type with Protobuf attributes. The JSON file serialization didn't implement its own as `Newtonsoft.Json` automatically had default handling of properties without needing any markup.

A similar story exist for MongoDB and Redis - both implement their own classes for cache entries as both (MongoDB with BSON, Redis with Protobuf) needed attributes on the type to support serialization.

I think the serializers will need to support caching raw value data (the value someone wants to cache), a `CacheEntry<T>` and a `ManifestEntry`. We may be able to accomplish this if the different serialization frameworks have other methods of defining how to serialize that isn't attributes. Like a custom `JsonConverter` for `Newtonsoft.Json` or `System.Text.Json` and whatever is the same for Protobuf.

Reply:
Do you consider this a blocker for this PR? I took a couple of swings at it but it feels like a pretty significant refactor unless the same approach is employed for Redis as is used for the file system layer.

Reply:
What I'll do is merge this PR in as-is and work on addressing the more repo-wide concerns in future PRs etc. This gets a core piece in where the RedisCacheLayer now supports both JSON and Protobuf serializers, even if other cache layers are still limited.